### PR TITLE
refactor: Move to package imports

### PR DESCRIPTION
**Pull request by spydon**
# Description

Moves from relative to package imports.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples`.

## Breaking Change

- [x] No, this is *not* a breaking change.

## Related Issues

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

**Comment by st-pasha** [comment]
This seems like it would be a big breaking change for any project that uses flame_lint.

**Comment by spydon** [comment]
> This seems like it would be a big breaking change for any project that uses flame_lint.

Is it a breaking change when it is only a warning? 🤔

**Comment by erickzanardo** [comment]
> Is it a breaking change when it is only a warning? 🤔


I agree, this is a dev dependency too, so nothing to worry IMO

**Comment by st-pasha** [comment]
> Is it a breaking change when it is only a warning?

When we put a `@Deprecated` annotation on a method or class, it creates warnings to whoever uses that method or class. And we don't consider deprecation to be a breaking change. So, it makes sense to say that warnings are not breaking change.

On the other hand, this change would cause hundreds if not thousands warnings at once, so one could argue that the sheer quantity of warnings causes this to become a "breaking change".

On an unrelated note, what is the reason for the switch? It seems to me that the vast majority of import statements are now longer. Also, the Dart itself seems to prefer relative imports (https://dart.dev/guides/language/effective-dart/usage#prefer-relative-import-paths).

File: packages/flame/test/collisions/collision_callback_benchmark.dart [hunk 1]
```diff
@@ -5,7 +5,7 @@ import 'package:flame/components.dart';
 import 'package:flame_test/flame_test.dart';
 import 'package:test/test.dart';
 
-import 'collision_test_helpers.dart';
+import '/home/spydon/repos/flame/packages/flame/test/collisions/collision_test_helpers.dart';
```
Comment (st-pasha) [inline packages/flame/test/collisions/collision_callback_benchmark.dart:8]: This doesn't seem right.

Comment (st-pasha) [reply packages/flame/test/collisions/collision_callback_benchmark.dart:8]: (also same in a few files below this one)

Comment (spydon) [reply packages/flame/test/collisions/collision_callback_benchmark.dart:8]: Good catch, will fix them.

**Comment by spydon** [comment]
> When we put a `@Deprecated` annotation on a method or class, it creates warnings to whoever uses that method or class. And we don't consider deprecation to be a breaking change. So, it makes sense to say that warnings are not breaking change.
> On the other hand, this change would cause hundreds if not thousands warnings at once, so one could argue that the sheer quantity of warnings causes this to become a "breaking change".

We will bump `flame_lint` just the same as it would have been a breaking change, but I don't want to put the PR as breaking since it touches all packages.

> On an unrelated note, what is the reason for the switch? It seems to me that the vast majority of import statements are now longer. Also, the Dart itself seems to prefer relative imports (https://dart.dev/guides/language/effective-dart/usage#prefer-relative-import-paths).

Among Flutter packages and applications it seems to be more standard to use only package imports only since it is easier to read when you have barrel files (no need for `../../../../..`.

**Review by erickzanardo — APPROVED** [review]

**Review by wolfenrain — APPROVED** [review]

**Pull request merged**